src/librustc_mir/borrow_check.rs

```diff
@@ -444,13 +455,14 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> MirBorrowckCtxt<'c, 'b, 'a, 'gcx, 'tcx>
                      context: Context,
                      lvalue_span: (&Lvalue<'tcx>, Span),
                      kind: (ShallowOrDeep, ReadOrWrite),
-                     flow_state: &InProgress<'b, 'gcx, 'tcx>) {
+                     flow_state: &InProgress<'b, 'gcx, 'tcx>) -> bool {
 
         let (sd, rw) = kind;
 
         // Check permissions
         self.check_access_permissions(lvalue_span, rw);
 
+        let mut has_storage_drop_or_dead_error_reported = false;
```

Comment:
Nit: we should probably just call this `error_reported`

Reply:
Resolved this, thanks.